docs/design/rubygems.md

```diff
@@ -0,0 +1,533 @@
+To avoid confusing users if a corner case shows up, we can add a `WARNING` log in case extra platforms are detected,
+and also document it properly. 
+
+Proper support for pre-compiled binaries should be probably left as a follow-up feature, similarly to what was done
+with [pip wheels](https://github.com/containerbuildsystem/cachi2/blob/main/docs/pip.md#distribution-formats).
+
+#### Checksum verification
+Since checksums in the `Gemlock.file` is still a feature in development (see [checksums](#dependency-checksums)), we
+can postpone implementing support for it until the feature is delivered.
+
+We need to decide if we will report all dependencies as having missing checksums in the SBOM, or not.
+
+#### Dev dependencies
+Bundler declares all dev dependencies under the `:development`
+[group](#dependency-groups-or-how-bundler-deals-with-dev-dependencies). Unfortunately, groups declared in the `Gemfile`
+are not reflected in the `Gemfile.lock`.
+
+To implement proper reporting of dev dependencies, we'll very likely need to also parse the `Gemfile`. It can be done
+as a follow-up if the need arises.
+
+#### Prefetching Bundler
+When running `bundle install`, Bundler will always try to fetch the exact version that is pinned in the `Gemfile.lock`
+to perform the install. When doing an offline install from cache, a warning message is instead printed, but Bundler
+will usually perform the install as expected.
+
+To allow users to use the pinned version instead of only relying on the Bundler version present in the base image,
+Cachi2 could also prefetch the specific Bundler version needed for that project. This is easy to achieve, since Bundler
+is treated as an ordinary Gem: https://rubygems.org/gems/bundler. 
+
+This feature, however, is out of scope for the initial implementation, and could be added if there's user demand for
+it.
+
+### Providing the content for the hermetic build
+
+#### Setting the Bundler configuration
+
+The order of precedence for Bundler configuration options is as follows:
+
+1. Local config (`<project_root>/.bundle/config or $BUNDLE_APP_CONFIG/config`)
+2. Environment variables (ENV)
+3. Global config (`~/.bundle/config`)
+4. Bundler default config
+
+Since the local configuration takes higher precedence than the environment variables (except `BUNDLE_APP_CONFIG`), we
+need to set the Bundler configuration options to make the build work.
+
+In order to do this, we can either use `inject-files` to overwrite the `.bundle/config` directory in the source folder,
+or use `BUNDLE_APP_CONFIG` to point Bundler to a config directory within the Cachi2 output directory. The latter has
+the benefit of not needing to dirty the cloned sources, but it wouldn't be able to support a multiple Ruby project per
+repository scenario (since we would need to keep multiple configuration files).
+
+#### Relevant configuration for the build
+
+```
+BUNDLE_CACHE_PATH=${output_dir}/deps/rubygems
+BUNDLE_DEPLOYMENT=true
+BUNDLE_FORCE_RUBY_PLATFORM=true
```

Comment:
From the tests I did, I didn't need to set this in order for the hermetic build to work properly. Can you confirm, @slimreaper35?

Reply:
I believe we don't need `BUNDLE_FORCE_RUBY_PLATFROM` anymore. It just forces bundler to download gems without native extensions. But we download everything manually... I tried to remove the env variable and the `ruby` from `PLATFORMS`, and the build works.